contracts/Strategy.sol

```diff
@@ -507,16 +508,26 @@ contract Strategy is BaseStrategy, DydxFlashloanBase, ICallee {
         //This part makes sure our withdrawal does not force us into liquidation
         (uint256 depositBalance, uint256 borrowBalance) = getCurrentPosition();
 
-        uint256 AmountNeeded = 0;
-        if(collateralTarget > 0){
-            AmountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
-        }
-        uint256 redeemable = depositBalance.sub(AmountNeeded);
+        uint256 amountNeeded = 0;
+        if(collateralTarget == 0){
+            collateralTarget = 1e15; // 0.001 * 1e18. lower we have issues
+        } 
+        
+        amountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
 
-        if (redeemable < _amount) {
-            cToken.redeemUnderlying(redeemable);
-        } else {
-            cToken.redeemUnderlying(_amount);
+        if(depositBalance >= amountNeeded){
+            uint256 redeemable = depositBalance.sub(amountNeeded);
```

Comment:
are you sure that `depositBalance > amountNeeded `. This is one of the issues I had yesterday.

Reply:
what do you mean?

Reply:
I was getting a revert when testing matt code and `amountNeeded` was greater than `depositBalance`

Reply:
i am still not sure what you mean? we are testing here is amountNeeded  is greater than depositBalance